### ISSUE TITLE
Clarification for usage of negative loss with `torch.optim.lr_scheduler.ReduceLROnPlateau`

### ISSUE THREAD

**Issue by yks72p** [pytorch/pytorch#38622]
## 📚 Documentation

It would be nice to have explicit statement in the documentation that `torch.optim.lr_scheduler.ReduceLROnPlateau` with `mode=min` and `threshold_mode='rel'` expects loss to be `> 0`

This is the excerpt from documentation about `threshold_mode`:
```
threshold_mode (str): One of `rel`, `abs`. In `rel` mode,
dynamic_threshold = best * ( 1 + threshold ) in 'max'
mode or best * ( 1 - threshold ) in `min` mode.
In `abs` mode, dynamic_threshold = best + threshold in
`max` mode or best - threshold in `min` mode. Default: 'rel'.
```

And this is the corresponding excerpt from the code:
```python
def is_better(self, a, best):
    if self.mode == 'min' and self.threshold_mode == 'rel':
        rel_epsilon = 1. - self.threshold
        return a < best * rel_epsilon

    elif self.mode == 'min' and self.threshold_mode == 'abs':
        return a < best - self.threshold

    elif self.mode == 'max' and self.threshold_mode == 'rel':
        rel_epsilon = self.threshold + 1.
        return a > best * rel_epsilon

    else:  # mode == 'max' and epsilon_mode == 'abs':
        return a > best + self.threshold
```

I use `NegativeDiceLoss` in my experiments so the loss is negative and is expected to be minimized. The only correct mode of `ReduceLROnPlateau` I can use is `threshold_mode='abs'` because `threshold_mode='rel'` will not work for minimizing negative loss.

Imagine `mode='min', threshold_mode='rel', threshold=0.1`:
* If we use loss that is expected to be > 0 all the time and current loss value is `10` than dynamic threshold will be `(1 - 0.1) * 10 == 9`. And the check will be `a < 9`. So there will be a margin between previous loss value and the new one to ensure that model improved.
* If we use loss that is expected to be < 0 all the time (as in my case with `NegativeDiceLoss`) and current loss value is `-10` than dynamic threshold will be `(1 - 0.1) * -10 == -9`. And the check will be `a < -9`. __It does not correspond to logic of scheduler__ as it allows new loss value to be slightly worse than previous. As I understand the goal of the dynamic threshold is to guarantee that the loss improved __by some margin__ to deal with noise in loss calculations.

If my understanding is correct it is better to __explicitly state in documentation__ that for `threshold_mode='rel'` only positive loss functions are expexted. Probably negative loss functions are not common to use but sometimes they are required (as with dice score as loss function).
Please correct me if I'm wrong :)

**Comment by zuzhaoye** [comment]
Exactly. I found the same issue when dealing with reinforcement learning, and passing the negative rewards to ReduceLROnPlateau. Thank you @yks72p for pointing this out and @agolynski for documenting it.

**Comment by apple2373** [comment]
I just had this issue with a negative loss value. Thanks for pointing this out. This issue helped me a lot, and I agree that pytorch should document it or shows the warning when negative loss is passed.
